src/DynamoCore/Core/PropertyChangeManager.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Dynamo.Core
+{
+    /// <summary>
+    /// Class can be used to stop property change notifications from being raised if the client object
+    /// inherits from <see cref="NotificationObject"/> and uses RaisePropertyChanged to raise prop notifications.
+    /// Use like:
+    /// using (node.SetPropsToSupress("frozen")){
+    /// node.frozen = true;
+    /// }
+    /// </summary>
+    internal class PropertyChangeManager : IDisposable
+    {
+        //TODO lock field during these accessors or use concurrent structure?
+        protected IEnumerable<string> PropertyNames { get; set; }
+        /// <summary>
+        /// Disable property change notifications for a list of a props.
+        /// The previous set of suppressed notifications will be cleared.
+        /// This API is meant to be used only for a short time and Dispose should be
+        /// called on the object this call returns.
+        /// </summary>
+        /// <param name="props"></param>
+        /// <returns></returns>
+        public PropertyChangeManager SetPropsToSuppress(params string[] props)
+        {
+            PropertyNames = props;
+            return this;
+        }
+        /// <summary>
+        /// Disposing this class will allow prop notifications to be raised.
+        /// </summary>
+        public void Dispose()
+        {
+            PropertyNames = null;
```

Comment:
Do we need to trigger the property change notification on dispose?
If we do not...then may e we should specify it in the comments

Reply:
No, we do not want to trigger the updates on dispose - or at least that will not make this class useful for the problem I am trying to solve with how I am currently using it.

Reply:
Sounds like we could improve this class in the future to also have a method called `SetPropsToDelayNotification` that could only trigger the properties that were delayed (on dispose).